ui/src/loading.test.tsx

```diff
@@ -0,0 +1,36 @@
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import React from "react";
+import Loading from "./loading";
+
+test("loading", async () => {
+  const reload = jest.fn();
+  const props = { isPending: true, reload: reload };
+  const loading = () => <Loading status={props}>loaded</Loading>;
+
+  const { rerender } = render(loading());
+  await waitFor(() =>
+    expect(screen.getByRole("progressbar")).toBeInTheDocument()
```

Comment:
you don't need to use waitFor if you replace getByRole with findByRole.  not sure if you just find this more clear.

also the "toBeInTheDocument" seems maybe recommended, but doesn't matter.  ie the entire thing could be replace by:
await screen.findByRole("progressbar");

Reply:
Nice, that's much shorter.

Reply:
Nice, that's much shorter.